facia-json/src/main/scala/com/gu/facia/client/ApiClient.scala

```diff
@@ -1,37 +1,82 @@
+import com.gu.etagcaching.FreshnessPolicy.AlwaysWaitForRefreshedValue
+import com.gu.etagcaching.aws.s3.{ObjectId, S3ByteArrayFetching}
+import com.gu.etagcaching.fetching.Fetching
+import com.gu.etagcaching.{ConfigCache, ETagCache}
+import com.gu.facia.client.models.{CollectionJson, ConfigJson}
 import play.api.libs.json.{Format, Json}
 
 import scala.concurrent.{ExecutionContext, Future}
 
-object ApiClient {
-  val Encoding = "utf-8"
+trait ApiClient {
+  def config: Future[ConfigJson]
+
+  def collection(id: String): Future[Option[CollectionJson]]
 }
 
-case class ApiClient(
+object ApiClient {
+  private val Encoding = "utf-8"
+
+  /**
+   * Legacy constructor for creating a client that does not support caching. Use `ApiClient.withCaching()` instead.
+   */
+  def apply(
     bucket: String,
-    /** e.g., CODE, PROD, DEV ... */
-    environment: String,
+    environment: String, // e.g., CODE, PROD, DEV ...
     s3Client: S3Client
-)(implicit executionContext: ExecutionContext) {
-  import com.gu.facia.client.ApiClient._
-
-  private def retrieve[A: Format](key: String): Future[Option[A]] = s3Client.get(bucket, key) map {
-    case FaciaSuccess(bytes) =>
-        Some(Json.fromJson[A](Json.parse(new String(bytes, Encoding))) getOrElse {
-          throw new JsonDeserialisationError(s"Could not deserialize JSON in $bucket, $key")
-        })
-    case FaciaNotAuthorized(message) => throw new BackendError(message)
-    case FaciaNotFound(_)  => None
-    case FaciaUnknownError(message)  => throw new BackendError(message)
+  )(implicit executionContext: ExecutionContext): ApiClient = new ApiClient {
+    val env: Environment = Environment(environment)
+
+    private def retrieve[A: Format](key: String): Future[Option[A]] = s3Client.get(bucket, key).map(translateFaciaResult[A](_))
+
+    def config: Future[ConfigJson] =
+      retrieve[ConfigJson](env.configS3Path).map(getOrWarnAboutMissingConfig)
+
+    def collection(id: String): Future[Option[CollectionJson]] =
+      retrieve[CollectionJson](env.collectionS3Path(id))
+  }
+
+  private def getOrWarnAboutMissingConfig(configOpt: Option[ConfigJson]): ConfigJson =
+    configOpt.getOrElse(throw BackendError("Config was missing!! NOT GOOD!"))
+
+  private def translateFaciaResult[B: Format](faciaResult: FaciaResult): Option[B] = faciaResult match {
+    case FaciaSuccess(bytes) => Some(parseBytes(bytes))
+    case FaciaNotAuthorized(message) => throw BackendError(message)
+    case FaciaNotFound(_) => None
+    case FaciaUnknownError(message) => throw BackendError(message)
   }
 
-  def config: Future[ConfigJson] =
-    retrieve[ConfigJson](s"$environment/frontsapi/config/config.json").map(_ getOrElse {
-      throw new BackendError("Config was missing!! OH MY GOD")
-    })
+  private def parseBytes[B: Format](bytes: Array[Byte]): B =
+    Json.fromJson[B](Json.parse(new String(bytes, Encoding))) getOrElse {
+      throw JsonDeserialisationError(s"Could not deserialize JSON")
```

Comment:
It might be nice to log out the actual error here - maybe one for a future pr? @rtyley